### PR TITLE
Document partitioner_ui.md: add ideas from the mail thread

### PR DESCRIPTION
**Pull request by ancorgs**
We have discussed several ideas about the future of the YaST Partitioner in this [mail thread](https://lists.opensuse.org/yast-devel/2020-04/msg00062.html).

This incorporates all those ideas to the document we are using as a base for the discussion, so all ideas can be reviewed together to move from there.

Better review using the "view file" link in the `partitioner_ui.md` file. That will show a rendered version of the document that can be navigated. The only section with new content is "Ideas", you can skip the rest of the document.

### CODE REVIEW

File: doc/partitioner_ui/ideas/leaf_nodes.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+course, they could also decide to delete or resize it, which will result in a new free space
+appearing in the list.
+
+If they select the free space in `vg0`, they would have the option to create a new logical volume.
+
+If they select `sdc`, they would be able to format it or to create a partition table on it. Maybe
+even the option to use it as physical volume for an existing or new volume group.
+
+In short, the main point is to get a curated list of devices that the users likely want to modify
+in your their next action and to offer the applicable/reasonable options for each of them.
```
Comment (dgdavid) [inline doc/partitioner_ui/ideas/leaf_nodes.md:66]: "your their". Maybe "their" or just "the"

File: doc/partitioner_ui/ideas/diagram.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+### Drawbacks
+
+This idea has an obvious drawback: such graphical view is very hard (if possible) to translate to
+the ncurses interface.
+
+It's also not so clear how well will it scale for complex scenarios. For SLE customers it's
+relatively common to have systems with 10+ disks, multipath, RAID and LVM on top. A visual approach
+could be harder to manage in such scenarios than something based in tables or similar widgets,
+although maybe the idea of increasing/decreasing the detail with the zoom level can help in that
+regard. On one hand, users with very complex storage setups often use the ncurses interface.
```
Comment (dgdavid) [inline doc/partitioner_ui/ideas/diagram.md:37]: I'm not sure, but maybe "On the other hand" ?

File: doc/partitioner_ui/ideas/leaf_nodes.md (hunk 1 above)
Comment (ancorgs) [reply doc/partitioner_ui/ideas/leaf_nodes.md]: fixed

File: doc/partitioner_ui/ideas/diagram.md (hunk 2 above)
Comment (ancorgs) [reply doc/partitioner_ui/ideas/diagram.md]: fixed

**Pull request merged**